modules/git/git.go

```diff
@@ -119,6 +119,9 @@ func Init(ctx context.Context) error {
 	if err := checkAndSetConfig("core.quotePath", "false", true); err != nil {
 		return err
 	}
+	if err := checkAndSetConfig("receive.advertisePushOptions", "true", true); err != nil {
```

Comment:
This doesn't work with anything older than ~2.13, might need to lock it behind check and add info to docs about that limitation.

Reply:
Ah, shoot good catch. I even read that, but for some reason thought we were requiring above that.  
Now I see those other features are also optional depending on the Git version.

Reply:
Done. According to GitLab's documentation, it requires Git >= 2.10